packages/amplify-graphql-api-construct/src/__tests__/__functional__/stack.test.ts

```diff
@@ -0,0 +1,35 @@
+import * as cdk from 'aws-cdk-lib';
+import { AmplifyGraphqlApi } from '../../amplify-graphql-api';
+import { AmplifyGraphqlDefinition } from '../../amplify-graphql-definition';
+
+describe('stack parameter access and verification', () => {
+  it('allows access to stack parameter', () => {
+    const api = new AmplifyGraphqlApi(new cdk.Stack(), 'TestApi', {
+      definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
+        type Todo @model @auth(rules: [{ allow: public }]) {
+          description: String!
+        }
+      `),
+      authorizationModes: {
+        apiKeyConfig: { expires: cdk.Duration.days(7) },
+      },
+    });
+
+    expect(api.stack).toBeDefined();
+  });
+
+  it('verifies the stack property is set to the expected value', () => {
+    const api = new AmplifyGraphqlApi(new cdk.Stack(), 'TestApi', {
+        definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
+            type Todo @model @auth(rules: [{ allow: public }]) {
+            description: String!
+            }
+        `),
+        authorizationModes: {
+            apiKeyConfig: { expires: cdk.Duration.days(7) },
+        },
+    });
+
+    expect(api.stack).toBe(cdk.Stack.of(api));
```

Comment:
```suggestion
  it('verifies the stack property is set to the expected value', () => {
    const stack = new cdk.Stack();
    const api = new AmplifyGraphqlApi(stack, 'TestApi', {
        definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
            type Todo @model @auth(rules: [{ allow: public }]) {
            description: String!
            }
        `),
        authorizationModes: {
            apiKeyConfig: { expires: cdk.Duration.days(7) },
        },
    });

    expect(api.stack).toBe(stack);
```


I think the first test is redundant with this one. `expect(api.stack).toBe(stack);` already asserts that prop is defined.


An idea for additional test would be something like this. (pseudocode)

```

class CustomL3Construct extends Construct {
 data: DataConstruct;
 constructor ( scope, ..whatever else here) {
   this.data = new DataConstruct(scope);
 }

}

// and then in test

const stack = new Stack();
const customConstruct = new CustomL3Construct(stack);

assert( customConstruct.data.stack).toBe(stack);
```

I.e. assert that we traverse tree upwards from data construct to find nearest parent stack.

Reply:
Would the suggested test be something that might make more sense in the `amplify-backend` tests? Since we would have an easier time traversing the stack tree in there

Reply:
> Would the suggested test be something that might make more sense in the amplify-backend tests?

for auth, function and storage yes. We have not explicitly tested that parent stack might be couple of levels up. But for data such test should be here.

On the other hand we put some trust in `Stack.of` API to do the right thing, so maybe it would be too paranoid. I.e. should we test that CDK works? Feel free to ignore that suggestion.

Reply:
I think it would be a good idea, making the CustomL3Construct is what I am a little unsure about, I'll look around and see if something like that has already been made

Reply:
> making the CustomL3Construct is what I am a little unsure about

It would be something that lives only in the test and can be empty construct. So that we can test stack->some_construct->data_construct nesting.

This might be helpful resource https://docs.aws.amazon.com/cdk/v2/guide/constructs.html .